spyder/app/mainwindow.py

```diff
@@ -673,49 +692,59 @@ def create_edit_action(text, tr_text, icon):
         self.file_menu = self.menuBar().addMenu(_("&File"))
         self.file_toolbar = self.create_toolbar(_("File toolbar"),
                                                 "file_toolbar")
-
+        self.file_menu.aboutToShow.connect(self.hide_menus)
         # Edit menu/toolbar
         self.edit_menu = self.menuBar().addMenu(_("&Edit"))
         self.edit_toolbar = self.create_toolbar(_("Edit toolbar"),
                                                 "edit_toolbar")
-
+        self.edit_menu.aboutToShow.connect(self.hide_menus)
         # Search menu/toolbar
         self.search_menu = self.menuBar().addMenu(_("&Search"))
         self.search_toolbar = self.create_toolbar(_("Search toolbar"),
                                                     "search_toolbar")
-
+        self.search_menu.aboutToShow.connect(self.hide_menus)
         # Source menu/toolbar
         self.source_menu = self.menuBar().addMenu(_("Sour&ce"))
         self.source_toolbar = self.create_toolbar(_("Source toolbar"),
                                                     "source_toolbar")
-
+        self.source_menu.aboutToShow.connect(self.hide_menus)
         # Run menu/toolbar
         self.run_menu = self.menuBar().addMenu(_("&Run"))
         self.run_toolbar = self.create_toolbar(_("Run toolbar"),
                                                 "run_toolbar")
+        self.run_menu.aboutToShow.connect(self.hide_menus)
 
         # Debug menu/toolbar
         self.debug_menu = self.menuBar().addMenu(_("&Debug"))
         self.debug_toolbar = self.create_toolbar(_("Debug toolbar"),
                                                     "debug_toolbar")
+        self.debug_menu.aboutToShow.connect(self.hide_menus)
 
         # Consoles menu/toolbar
         self.consoles_menu = self.menuBar().addMenu(_("C&onsoles"))
         self.consoles_menu.aboutToShow.connect(
                 self.update_execution_state_kernel)
 
+        self.consoles_menu.aboutToShow.connect(self.hide_menus)
+
         # Projects menu
         self.projects_menu = self.menuBar().addMenu(_("&Projects"))
         self.projects_menu.aboutToShow.connect(self.valid_project)
+        self.projects_menu.aboutToShow.connect(self.hide_menus)
+
         # Tools menu
         self.tools_menu = self.menuBar().addMenu(_("&Tools"))
+        self.tools_menu.aboutToShow.connect(self.hide_menus)
 
         # View menu
         self.view_menu = self.menuBar().addMenu(_("&View"))
+        self.view_menu.aboutToShow.connect(self.hide_menus)
 
         # Help menu
         self.help_menu = self.menuBar().addMenu(_("&Help"))
-
+        self.help_menu.aboutToShow.connect(self.hide_menus)
+        
+        self.setup_menus()

```

Comment:
@juanis2112 we could then add all menus to a variable like

```python
self.menus = [self.file_menu, .... , self.help_menu]
```

And then use this on the method like:

```python
for menu in self.menus:
   ...
```